File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/InferenceClassification.java
package graphjavav4.utilities.users;

import java.util.Objects;
import graphjavav4.utilities.users.inferenceClassification.InferenceClassificationOverride;
import com.microsoft.kiota.serialization.SerializationWriter;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.Parsable;
import java.util.function.BiConsumer;
import java.util.Map;
import java.util.HashMap;
import java.util.List;
public class InferenceClassification extends Entity implements Parsable {
    @javax.annotation.Nullable
    public List<InferenceClassificationOverride> overrides;
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeCollectionOfObjectValues("overrides", overrides);
    }
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(super.getDeserializeFields());
        fields.put("overrides", (o, n) -> { ((InferenceClassification)o).overrides = n.getCollectionOfObjectValues(InferenceClassificationOverride.class); });
        return fields;
    }
}

File: msgraph-mail/go/utilities/models/inference_classification.go
package models

import (
    i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55 "github.com/microsoft/kiota/abstractions/go/serialization"
)

type InferenceClassification struct {
    Entity
    overrides []InferenceClassificationOverride;
}
func NewInferenceClassification()(*InferenceClassification) {
    return nil
}
func (m *InferenceClassification) getOverrides()([]InferenceClassificationOverride) {
    return m.overrides
}
func (m *InferenceClassification) getFieldDeserializers()(map[string]func(interface{}, i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.ParseNode)(error), error) {
    return nil, nil
}
func (m *InferenceClassification) serialize(writer i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.SerializationWriter)(error) {
    return nil
}
func (m *InferenceClassification) setOverrides(value []InferenceClassificationOverride)() {
    m.overrides = value
}
